lib/krane/kubernetes_resource/stateful_set.rb

```diff
@@ -22,26 +22,16 @@ def status
     def deploy_succeeded?
       success = observed_generation == current_generation
 
-      @pods.each do |pod|
-        success &= pod.definition["metadata"]["labels"]["controller-revision-hash"] == status_data['updateRevision']
-      end
-
-      if update_strategy == 'RollingUpdate'
-        success &= status_data['currentRevision'] == status_data['updateRevision']
-        success &= desired_replicas == status_data['readyReplicas'].to_i
-        success &= desired_replicas == status_data['currentReplicas'].to_i
-
-      elsif update_strategy == ONDELETE
+      if update_strategy == ONDELETE && required_rollout != "full"
         unless @success_assumption_warning_shown
-          @logger.warn("WARNING: Your StatefulSet's updateStrategy is set to OnDelete, "\
-                       "which means the deployment won't succeed until all pods are updated by deletion.")
+          @logger.warn("WARNING: Your StatefulSet's updateStrategy is set to #{update_strategy}, "\
+                       "which means updates will not be applied until its pods are deleted.")
           @success_assumption_warning_shown = true
         end
-
-        if required_rollout == 'full'
-          success &= desired_replicas == status_data['readyReplicas'].to_i
-          success &= desired_replicas == status_data['updatedReplicas'].to_i
-        end
+      else
+        success &= status_data['currentRevision'] == status_data['updateRevision']
```

Comment:
Should we explicitly check for RollingUpdate when requiring this condition? If I'm understanding correctly, this check would currently still be applied to OnDelete where `required_rollout == "full"`, but as you [mentioned above](https://github.com/Shopify/krane/pull/950/files#r1548064091), `currentRevision` [isn't reliably set](https://github.com/kubernetes/kubernetes/issues/106055) for the OnDelete strat